Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs

```diff
@@ -164,6 +204,38 @@ private PropertyViewModel GetViewModel (IPropertyInfo property)
 			return new StringPropertyViewModel (property, this.editors);
 		}
 
+		void FilterData ()
+		{
+			this.properties.Clear ();
+
+			UpdateProperties ();
+		}
+
+		List<PropertyViewModel> cachedProperties;
+		public List<PropertyViewModel> CachedProperties {
+			get {
+				if (cachedProperties == null) {
+					cachedProperties = Properties.ToList ();
+				}
+				return cachedProperties;
+			}
+		}
+
+		IEnumerable<IGrouping<string, PropertyViewModel>> cachedPropertiesGroupBy;
+		public IEnumerable<IGrouping<string, PropertyViewModel>> CachedPropertiesGroupBy {
+			get {
+				if (cachedPropertiesGroupBy == null) {
+					cachedPropertiesGroupBy = CachedProperties.GroupBy (arg => arg.Category);
```

Comment:
This is just a cache of the lazy grouping, it's going to regroup each enumeration of it anyway, we need to capture the results into a list.